external-api/src/v1/controllers/number-generator.controller.ts

```diff
@@ -1,85 +1,92 @@
+import { ENTITY_TYPE, NUMBER_TYPE, USER } from '../../constants';
 dotenv.config();
-const numberGeneratorFunctionUrl = process.env.AZURE_NUMBER_GENERATOR_FUNCTION_URL;
 
-const callNumberGenerator = async ({ dealType, entityType, entityId, dealId, user }: any) => {
-  const response: any = await axios({
-    method: 'POST',
-    url: `${numberGeneratorFunctionUrl}/api/orchestrators/numbergenerator`,
-    data: {
-      dealType,
-      entityType,
-      entityId,
-      dealId,
-      user,
-    },
-  }).catch((error: any) => ({
-    error,
-  }));
+const { APIM_MDM_URL, APIM_MDM_VALUE, APIM_MDM_KEY } = process.env;
 
-  if (response.error) {
-    await durableFunctionsLogController.addDurableFunctionLog({
-      type: 'NUMBER_GENERATOR',
-      data: {
-        dealType,
-        entityType,
-        entityId,
-        dealId,
-        user,
-        error: await response.error.toJSON(),
-      },
-    });
+const headers = {
+  headers: {
+    [String(APIM_MDM_KEY)]: APIM_MDM_VALUE,
+    'Content-Type': 'application/json',
+  },
+};
 
-    return {
-      status: response.status || 500,
-    };
+/**
+ * Determines the number type based on the entity type.
+ * @param entityType - The type of entity for which the number type needs to be determined.
+ * @returns A number based on the value of entityType.
+ * @throws {Error} If entityType is invalid.
+ */
+export const getNumberTypeId = (entityType: string): number => {
+  switch (entityType) {
+    case ENTITY_TYPE.DEAL:
+      return NUMBER_TYPE.DEAL;
+    case ENTITY_TYPE.FACILITY:
+      return NUMBER_TYPE.FACILITY;
+    default:
+      throw new InvalidEntityTypeError(entityType);
   }
+};
+
+/**
+ * Retrieves a number from a number generator API based on the provided `entityType` and `dealId`.
+ * @param req - The HTTP request object containing the `entityType` and `dealId` in the `body` property.
+ * @param res - The HTTP response object used to send the response back to the client.
+ * @returns {Promise<Object>} The retrieved number in the response body.
+ */
+export const get = async (req: Request, res: Response): Promise<Response<NumberGeneratorResponse> | Response<NumberGeneratorErrorResponse>> => {
+  try {
+    const { entityType, dealId } = req.body;
+    const numberTypeId = getNumberTypeId(entityType);
+    const endpoint = `${APIM_MDM_URL}numbers`;
+    const payload = {
+      numberTypeId,
+      createdBy: USER.DTFS,
+      requestingSystem: USER.DTFS,
+    };
 
-  const { id: instanceId, ...numberGeneratorFunctionUrls } = response.data;
+    console.info('⚡️ Invoking number generator for deal %s', dealId);
 
-  await durableFunctionsLogController.addDurableFunctionLog({
-    type: 'NUMBER_GENERATOR',
-    data: {
-      instanceId,
-      dealType,
-      entityType,
-      entityId,
-      dealId,
-      numberGeneratorFunctionUrls,
-      user,
-    },
-  });
+    const response: NumberGeneratorResponse = await axios.post(endpoint, [payload], headers);
```

Comment:
is payload wrapped in an array because you can post multiple entities at once?

Reply:
Currently MDM only accepts one at a time but that is the intention for future implementation in APIM.